service/storage_proxy.cc

```diff
@@ -330,6 +331,43 @@ class cas_mutation : public mutation_holder {
     }
 };
 
+// Tracks the success of a set of CDC mutations generated from the same mutation
```

Comment:
Is it really "tracks the success"? It sounds like a when_all_succeeds() could have done this, and here you have much more - _was_split, _touches and all sorts of other stuff... So I don't think I understand the purpose of this structure, or again (and I'm repeating myself...) why the code can't just update the statistics immediately when something happens (e.g., a split) instead of remembering _was_split.

Reply:
(Now this object is called `operation_result_tracker`)

Currently, all CDC counters come in two flavors: "total" and "failed", where "total" means a cumulative value for all operations performed, and "failed" means a value for all operations that failed.

We cannot determine in advance if writing cdc mutations will succeed or fail, so we need to store details about what was happening in the operation. When the operation fails, these details will be used to update relevant "failed" counters.